internal/runners/publish/publish.go

```diff
@@ -118,33 +127,54 @@ func (r *Runner) Run(params *Params) error {
 		reqVars.Name = filepath.Base(params.Filepath)
 	}
 
-	ts := time.Now()
-	ingredients, err := model.SearchIngredientsStrict(reqVars.Namespace, reqVars.Name, true, false, &ts)
-	if err != nil && !errs.Matches(err, &model.ErrSearch404{}) { // 404 means either the ingredient or the namespace was not found, which is fine
-		return locale.WrapError(err, "err_uploadingredient_search", "Could not search for ingredient")
+	var ingredient *ParentIngredient
+
+	isRevision := false
+	if params.Version != "" {
+		// Attempt to get the version if it already exists, it not existing is not an error though
+		i, err := model.GetIngredientByNameAndVersion(params.Namespace, params.Name, params.Version)
+		if err != nil {
+			return locale.WrapInputError(err, "err_uploadingredient_getversion", "Could not grab ingredient by version")
+		}
+		ingredient = &ParentIngredient{*i.IngredientID, *i.IngredientVersionID, *i.Version, i.Dependencies}
+		isRevision = true
+	}
+
+	if ingredient == nil {
+		// Attempt to find the existing ingredient, if we didn't already get it from the version specific call above
+		ts := time.Now()
```

Comment:
Do we want to use the latest platform timestamp here? Will using a timestamp that is potentially ahead of the platform cause issues?

Reply:
It shouldn't cause issues. It's essentially telling the platform "you can use ingredients as new as this date". So theoretically even a date 100 years in the future should work.

The platform timestamp wouldn't be appropriate here since that would predate the timestamp that a user published their own ingredient.